notification/app/notification/services/azure/NewsstandSender.scala

```diff
@@ -1,19 +1,40 @@
 package notification.services.azure
 
+import java.time.{Clock, Duration, Instant}
 import java.util.UUID
 
 import azure.{APNSRawPush, NotificationHubClient, Tags}
-import models.Topic
+import com.gu.notificationschedule.dynamo.{NotificationSchedulePersistenceAsync, NotificationsScheduleEntry}
+import models.{NewsstandShardConfig, Topic}
 import models.TopicTypes.Newsstand
 import notification.models.ios.NewsstandNotification
 
-class NewsstandSender(hubClient: NotificationHubClient) {
+import scala.concurrent.{ExecutionContext, Future}
 
+
+class NewsstandSender(
+                       hubClient: NotificationHubClient,
+                       legacyNewsstandRegistrationConverterConfig:NewsstandShardConfig,
+                       notificationSchedulePersistence: NotificationSchedulePersistenceAsync,
+                       clock: Clock = Clock.systemUTC()
+                     ) {
+  implicit  val executionContext:ExecutionContext = ExecutionContext.Implicits.global
+  val sevenDaysInSeconds = Duration.ofDays(7).getSeconds
   def sendNotification(id: UUID) = {
     val push = APNSRawPush(
       body = NewsstandNotification(id).payload,
       tags = Some(Tags.fromTopics(Set(Topic(Newsstand, "newsstand"))))
     )
-    hubClient.sendNotification(push)
+    val shards = legacyNewsstandRegistrationConverterConfig.shards
+    val nowSeconds = Instant.now(clock).getEpochSecond
+
+    hubClient.sendNotification(push).flatMap(hubResult => {
+      Future.sequence{Range(1,shards.toInt + 1).map(shard => notificationSchedulePersistence.writeAsync(NotificationsScheduleEntry(
+        new UUID(id.getMostSignificantBits, id.getLeastSignificantBits+shard).toString,
+        shard.toString,
```

Comment:
I think you may be missing the json serialisation of the notification, or have I missed something myself?

Reply:
Adding that too